### PR TITLE
Remove unnecessary slashing validation

### PR DESCRIPTION
**Pull request by jframe**
Change slashing protection validation

* So this only happens if slashing protection is enabled
* Type parameter is no longer required if slashing protection isn't enabled
* Default slashing protection enabled to false until slashing protection is finished

### CODE REVIEW

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/Eth2SignForIdentifierHandler.java [hunk 1]
```diff
@@ -71,8 +69,12 @@ public void handle(final RoutingContext routingContext) {
           .ifPresentOrElse(
               signature -> {
                 if (slashingProtection.isPresent()) {
-                  if (maySign(identifier, eth2SigningRequestBody)) {
-                    respondWithSignature(routingContext, signature);
+                  try {
+                    if (maySign(identifier, eth2SigningRequestBody)) {
+                      respondWithSignature(routingContext, signature);
+                    }
+                  } catch (IllegalArgumentException e) {
```
Comment (usmansaleem) [inline core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/Eth2SignForIdentifierHandler.java:76]: nit: final

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/Eth2SignForIdentifierHandler.java [hunk 2]
```diff
@@ -71,8 +69,12 @@ public void handle(final RoutingContext routingContext) {
           .ifPresentOrElse(
               signature -> {
                 if (slashingProtection.isPresent()) {
-                  if (maySign(identifier, eth2SigningRequestBody)) {
-                    respondWithSignature(routingContext, signature);
+                  try {
+                    if (maySign(identifier, eth2SigningRequestBody)) {
+                      respondWithSignature(routingContext, signature);
+                    }
```
Comment (usmansaleem) [inline core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/Eth2SignForIdentifierHandler.java:75]: should there be an "else" here? responding with appropriate `routingContext.fail(code)` ??

**Review by usmansaleem — APPROVED** [review]
LGTM - Minor comments.

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/Eth2SignForIdentifierHandler.java (hunk 1 above)
Comment (jframe) [reply core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/Eth2SignForIdentifierHandler.java]: done

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/Eth2SignForIdentifierHandler.java (hunk 2 above)
Comment (jframe) [reply core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/Eth2SignForIdentifierHandler.java:78]: done

**Pull request merged**